controlplane/kubeadm/controllers/scale.go

```diff
@@ -152,13 +133,6 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		}
 	}
 
-	// Do a final health check of the Control Plane components prior to actually deleting the machine
-	if err := r.managementCluster.TargetClusterControlPlaneIsHealthy(ctx, util.ObjectKey(cluster), kcp.Name); err != nil {
-		logger.V(2).Info("Waiting for control plane to pass control plane health check before removing a control plane machine", "cause", err)
-		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy",
-			"Waiting for control plane to pass control plane health check before removing a control plane machine: %v", err)
-		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
-	}
```

Comment:
These checks (in both scale up and scale down) were also gating the upgrade workflow as well and the new general check is only triggered during normal scale up/scale down operations currently.

Reply:
Good catch! So my new assumption is as far as the control plane is initialized, we want to run these health checks. Moved it before the upgrade.